src/pydata_sphinx_theme/toctree.py

```diff
@@ -427,7 +427,7 @@ def add_header_level_recursive(ul, level):
             if ul is None:
                 return
             if level <= (context["theme_show_toc_level"] + 1):
-                ul["class"] = [*ul.get("class", []), "visible"]
+                ul["class"] = [*ul.get("class", []), "pst-show_toc_level"]
```

Comment:
Would it be preferable to remove the `ul` elements that are nested beyond the config threshold from the HTML tree? Right now, we're styling all of the page TOC `ul`s with `display: none`, then applying a CSS class to the ones within the threshold that overrides the `display: none`.

Reply:
I would think so; it somehow feels like a better approach to only include `ul` elements within the threshold than adding everything and hiding them via `css`.
I am unsure if this approach has any rendering/generation implications, so maybe @drammock can shed some light?

Reply:
IIRC, the deeper TOC levels are *revealed* when their parent section is visible onscreen. So to me it makes more sense to handle those visibility changes by toggling a class on and off rather than deleting/re-adding the nodes every time the user scrolls in / out of a section.

Reply:
Ah this makes sense now!